qiskit/transpiler/passes/routing/swap_strategies/commuting_2q_block.py

```diff
@@ -0,0 +1,57 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A gate made of commuting two-qubit gates."""
+
+from typing import Set
+
+from qiskit import QiskitError
+from qiskit.circuit import Gate
+from qiskit.dagcircuit import DAGOpNode
+
+
+class Commuting2qBlocks(Gate):
+    """A gate made of commuting two-qubit gates.
+
+    This gate is intended for use with commuting swap strategies to make it convenient
+    for the swap strategy router to identify which blocks of operations commute.
+    """
+
+    def __init__(self, node_block: Set[DAGOpNode]):
+        """Initialize the op.
+
```

Comment:
Nothing blocking, but maybe just using `Iterable` is more general (would also need to be imported from `typing`)
```suggestion
    def __init__(self, node_block: Iterable[DAGOpNode]) -> None:
        """
```